Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -715,12 +716,22 @@ struct ZipBond {
 
   // bond a<->b for now only use single bonds
   //  XXX FIX ME take the highest bond order.
-  bool bond(RWMol &newmol) const {
+  bool bond(RWMol &newmol, const MolzipParams &params) const {
     if (!a || !b || !a_dummy || !b_dummy) {
       BOOST_LOG(rdWarningLog)
           << "Incomplete atom labelling, cannot make bond" << std::endl;
       return false;
     }
+    
+    // Fragment on bonds allows multiple links to the same atom
+    // i.e. C.[1C].[1C]
+    //  otherwise throw an invariant error
+    if (!(params.label == MolzipLabel::FragmentOnBonds) &&  a->getOwningMol().getBondBetweenAtoms(a->getIdx(), b->getIdx())) {
+      CHECK_INVARIANT(!a->getOwningMol().getBondBetweenAtoms(a->getIdx(), b->getIdx()),
+                      "molzip: zipped Bond already exists, perhaps labels are duplicated");
+    }
```

Comment:
wouldn't this work as:
```suggestion
    CHECK_INVARIANT(params.label == MolzipLabel::FragmentOnBonds || !a->getOwningMol().getBondBetweenAtoms(a->getIdx(), b->getIdx()),
                      "molzip: zipped Bond already exists, perhaps labels are duplicated");
```